### DISCUSSION TITLE
Different colors of tags (group)?

### DISCUSSION THREAD

**Discussion by elkrien** [qtile/qtile#2756] [Ideas] [answered]
Is it possible to set or if not to add the possibility to have active tags (group) colored with different colors? 
For example I have 5 tags (group), each of them are different symbols (icons). All of them are white when not active / selected. Now each active are colored with the same colors but would like to have different colors for each (i.e. tag 1 is red and tag 2 is blue...). Something like on this picture (left top corner)? - [Link](https://raw.githubusercontent.com/siduck76/dotfiles/master/rice%20flex/dashboard-nvim.png)

**Comment by m-col** [comment, marked as answer]
If you're specifically talking about the `GroupBox` widget then sure - but you'd need to subclass that widget in your config and specify the logic. You'll need to look in the `draw` method where the colour is specified for a given group when it loops over the groups. If you just override that method and use that widget, that should be enough.

**Reply by elkrien** [reply]
Any examples how to subclass that widget? Sorry if this is noob question but I'm still learning.

**Reply by m-col** [reply]
Basically it is creating a new widget (`ColouredGroupBox` below) with that widget as a base. For example:
```py
from libqtile import widget

class ColoredGroupBox(GroupBox):
    def draw(self):
        # Add custom code here
```
In that `draw` method you need to add your own code. All of the other methods are inherited by `GroupBox` and can be left untouched. Copy the implementation from the original `GroupBox.draw` method and then edit it to do what you want. There is a bit that deals with the text color and you just need to change that bit to set the text color for each group.

**Reply by elkrien** [reply]
Thank You very much.

EDIT:
I have found solution to manage this - by using `visible_groups` key. I can add multiple GroupBox widgets with different setting for each of them with only one item visible. Don't know if it is a correct way but it works ;)
Feel free to close this conversation.

**Reply by CreativeCodeCat** [reply]
> Thank You very much.
> 
> EDIT: I have found solution to manage this - by using `visible_groups` key. I can add multiple GroupBox widgets with different setting for each of them with only one item visible. Don't know if it is a correct way but it works ;) Feel free to close this conversation.

Btw i just created a widget in the qtile-extras to do this hope it gets added soon [#264](https://github.com/elParaguayo/qtile-extras/pull/264)

you could also just take that file and add it into a folder in the qtile config then import it until its added
